### PR TITLE
RAB-1125: [php 8.0 => 8.1] Fix breaking change

### PR DESCRIPTION
**Pull request by MadeWilson**
**Description (for Contributor and Core Developer)**

This Pr fix all breaking changes related to php 8.1, it does not do the upgrade yet has it will be done in a separate Pr.
Monolog has been upgraded has the current version throw a deprecation that cause a memory leak when building a dev env.
Ramsey has been upgraded has 3.x version contains a bug in php 8.1.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Message/ProductCreated.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ public function __construct(Author $author, array $data, int $timestamp = null,
         Assert::keyExists($data, 'identifier');
         Assert::nullOrString($data['identifier']);
         Assert::keyExists($data, 'uuid');
-        Assert::true(Uuid::isValid($data['uuid']));
+        Assert::true(Uuid::isValid($data['uuid']->serialize()));
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Enrichment/Component/Product/Message/ProductCreated.php:27]: I don't understand why we need to do that. If it's already a uuid we didn't need to serialize to check if it's an uuid as it's already an Uuid

Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Enrichment/Component/Product/Message/ProductCreated.php:27]:
```suggestion
        Assert::instanceOf($data['uuid'], UuidInterface::class);
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Message/ProductRemoved.php [hunk 2]
```diff
@@ -24,7 +24,7 @@ public function __construct(Author $author, array $data, int $timestamp = null,
         Assert::keyExists($data, 'identifier');
         Assert::nullOrString($data['identifier']);
         Assert::keyExists($data, 'uuid');
-        Assert::true(Uuid::isValid($data['uuid']));
+        Assert::true(Uuid::isValid($data['uuid']->serialize()));
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Enrichment/Component/Product/Message/ProductRemoved.php:27]: Same

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Settings/Persistence/DbalSelectConnectionWithCredentialsByCodeQuery.php [hunk 3]
```diff
@@ -74,7 +74,7 @@ public function execute(string $code): ?ConnectionWithCredentials
             $row['label'],
             $row['flow_type'],
             $row['image'],
-            $row['client_id'] . '_' . $row['random_id'],
+            $row['client_id'].'_'.$row['random_id'],
```
Comment (ghoussard) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Settings/Persistence/DbalSelectConnectionWithCredentialsByCodeQuery.php:77]:
```suggestion
            sprint('%s_%s', $row['client_id'], $row['random_id']),
```

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductUuidsIntegration.php [hunk 4]
```diff
@@ -141,13 +149,14 @@ public function test_that_it_gets_descendant_identifiers_of_root_product_models(
                 ],
             ]
         );
-        $aLargeRedShoe = $this->createProduct('a_large_red_shoe', 'familyA', 'a_large_shoe');
-        $aLargeGreenShoe = $this->createProduct('a_large_green_shoe', 'familyA', 'a_large_shoe');
+        $productUuids[] = $this->createProduct('a_large_red_shoe', 'familyA', 'a_large_shoe')->getUuid()->serialize();
+        $productUuids[] = $this->createProduct('a_large_green_shoe', 'familyA', 'a_large_shoe')->getUuid()->serialize();
+
+        $expectedUuids = array_map(static fn (UuidInterface $uuid) => $uuid->serialize(), $this->get('akeneo.pim.enrichment.product.query.get_descendant_variant_product_uuids')->fromProductModelCodes(['a_shirt', 'a_shoe']));
```
Comment (StevenVAIDIE) [inline tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductUuidsIntegration.php:155]:
```suggestion
        $expectedUuids = array_map(static fn (UuidInterface $uuid) => $uuid->toString(), $this->get('akeneo.pim.enrichment.product.query.get_descendant_variant_product_uuids')->fromProductModelCodes(['a_shirt', 'a_shoe']));
```

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductUuidsIntegration.php [hunk 5]
```diff
@@ -114,9 +119,12 @@ public function test_that_it_gets_descendant_identifiers_of_root_product_models(
                 ],
             ]
         );
-        $aMediumRedShirt = $this->createProduct('a_medium_red_shirt', 'familyA', 'a_medium_shirt');
-        $aMediumBlueShirt = $this->createProduct('a_medium_blue_shirt', 'familyA', 'a_medium_shirt');
-        $aLargeBlackShirt = $this->createProduct('a_large_black_shirt', 'familyA', 'a_large_shirt');
+
+        $productUuids = [];
+
+        $productUuids[] = $this->createProduct('a_medium_red_shirt', 'familyA', 'a_medium_shirt')->getUuid()->serialize();
```
Comment (StevenVAIDIE) [inline tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductUuidsIntegration.php:125]:
```suggestion
        $productUuids[] = $this->createProduct('a_medium_red_shirt', 'familyA', 'a_medium_shirt')->getUuid()->toString();
```

**Review by ghoussard — APPROVED** [review]
Would ask @akeneo/squad-chipmunks to review this PR pls ? Especially for changes concerning uuid usage within Enrichment

File: src/Akeneo/Pim/Enrichment/Component/Product/Message/ProductUpdated.php [hunk 6]
```diff
@@ -24,7 +24,7 @@ public function __construct(Author $author, array $data, int $timestamp = null,
         Assert::keyExists($data, 'identifier');
         Assert::nullOrString($data['identifier']);
         Assert::keyExists($data, 'uuid');
-        Assert::true(Uuid::isValid($data['uuid']));
+        Assert::true(Uuid::isValid($data['uuid']->serialize()));
```
Comment (StevenVAIDIE) [inline src/Akeneo/Pim/Enrichment/Component/Product/Message/ProductUpdated.php:27]:
```suggestion
        Assert::isInstanceOf($data['uuid'], UuidInterface::class);
```

**Review by StevenVAIDIE — APPROVED** [review]

**Review by mmetayer — APPROVED** [review]
GTM for Enrichment part

**Review by tseho — APPROVED** [review]

**Pull request merged**